### ISSUE TITLE
Underlying view's autolayout get's disturbed on exit

### ISSUE THREAD

**Issue by ericlewis** [Dimillian/DMCustomModalViewController#1]
First of all, i love it! I love it love it love it!

But, when utilizing storyboards, and autolayout- there seems to be a sizing issue that causes the app to relayout. will attach images.
![ios simulator screen shot jun 6 2013 11 33 54 am](https://f.cloud.github.com/assets/674503/619859/ab149462-ced7-11e2-8841-5638a7c35dac.png)
![ios simulator screen shot jun 6 2013 11 33 57 am](https://f.cloud.github.com/assets/674503/619860/ae050814-ced7-11e2-8b9b-6835937beb8e.png)
![ios simulator screen shot jun 6 2013 11 33 59 am](https://f.cloud.github.com/assets/674503/619862/afda4dc0-ced7-11e2-9999-0df5b66f3da0.png)

hopefully we can work it out! I found that by taking an SS of the underlying view, and animating THAT backwards solves most issues of layout.. I have been exploring my own versions- but your's is still the best. Let me know!

**Comment by ericlewis** [comment]
Adding [primaryView setNeedsDisplay]; seems to fix it for that view- but table views still become a different size..

**Comment by ericlewis** [comment]
It appears that it could be because of the size of my navbar being slightly different than the typical one..

**Comment by ericlewis** [comment]
one more note: it appears resetting the tableviews frame works on finish.

**Comment by Dimillian** [comment]
Hmm I never used it in production yet, I plan to integrate with my current app, so I'll do a massive debug. 
For what I know the animation I did suck, it reset the view everytime, + it don't work in landscape. 

If you can send me an example project I could start some debug on it :) I plan to totally redo the animation.

**Comment by ericlewis** [comment]
I can see if I can get a decent example together for you, from what i can tell it appears that my custom appearances (which are slightly taller image wise) may have an issue with tables. I solved it by resetting the frame, and subtracting 20 from the height on a successful hide. Landscape is something we don't even support at all- i think it would be a weird paradigm there. We'd love to use it in our Prod, and have made some tiny tweaks that seem to solve most of our problems. such as redraw on finish and reframing. I'll get you an example!

**Comment by ericlewis** [comment]
We are also using it in a tab view controller with table views inside, seems to work well. Our internal methods consisted of Screen shotting the parentview overlaying it on to a black bg subview, and manipulating that underneath the rootview, instead of the parentview scaling. but it lacked the integration with controllers that yours had.

**Comment by Dimillian** [comment]
Hmm sound nice, don't hesitate to integrate your fixes directly into DMCustomModal and do a pull reques :D

**Comment by ericlewis** [comment]
no problem! will do :)

**Comment by Dimillian** [comment]
That would be awesome, in some weeks I'll start using it in production. I use it actually, but my views still not complex. 
The thing is I'm doing everything in code, I've not tested it with Interface Builder nor Storyboarding, that why I need an example :)

**Comment by ericlewis** [comment]
yeah no problem, we are doing quick prototyping for a beta, which is essentially production- and plan to move to pure code gradually as we hire. maybe end of day, or tomorrow till i get you an example.

**Comment by shobhitkasliwal-zz** [comment]
any word on landscape support ?

**Comment by Dimillian** [comment]
@shobhitkasliwal I think I'll totally rewrite it for iOS 7 only.
